pkg/virt-controller/network/multus_annotations.go

```diff
@@ -56,18 +56,18 @@ func (mnap multusNetworkAnnotationPool) toString() (string, error) {
 	return string(multusNetworksAnnotation), nil
 }
 
-func GenerateMultusCNIAnnotation(namespace string, interfaces []v1.Interface, networks []v1.Network, config *virtconfig.ClusterConfig) (string, error) {
-	return GenerateMultusCNIAnnotationFromNameScheme(namespace, interfaces, networks, namescheme.CreateHashedNetworkNameScheme(networks), config)
+func GenerateMultusCNIAnnotation(namespace string, interfaces []v1.Interface, networks []v1.Network, networkToIPAMClaimParams map[string]IPAMClaimParams, config *virtconfig.ClusterConfig) (string, error) {
+	return GenerateMultusCNIAnnotationFromNameScheme(namespace, interfaces, networks, namescheme.CreateHashedNetworkNameScheme(networks), networkToIPAMClaimParams, config)
 }
 
-func GenerateMultusCNIAnnotationFromNameScheme(namespace string, interfaces []v1.Interface, networks []v1.Network, networkNameScheme map[string]string, config *virtconfig.ClusterConfig) (string, error) {
+func GenerateMultusCNIAnnotationFromNameScheme(namespace string, interfaces []v1.Interface, networks []v1.Network, networkNameScheme map[string]string, networkToIPAMClaimParams map[string]IPAMClaimParams, config *virtconfig.ClusterConfig) (string, error) {
```

Comment:
nit: can you find a way to refactor this function so the current usages do not need to be updated ? 

maybe entertain the idea of injecting the function as a parameter, and export the common part into an helper ? Something like:
```golang
func GenerateMultusCNIAnnotationFromNameScheme(namespace string, interfaces []v1.Interface, networks []v1.Network, networkNameScheme map[string]string, config *virtconfig.ClusterConfig) (string, error) {

    return commonMultusAnnotationGenerationFunc(
        namespace,
        interfaces,
        networks,
        networkNameScheme,
        config, 
        newMultusAnnotationData
    )
}

func GenerateMultusCNIAnnotationWithIPAMClaimsFromNameScheme(namespace string, interfaces []v1.Interface, networks []v1.Network, networkNameScheme map[string]string, config *virtconfig.ClusterConfig) (string, error) {

    return commonMultusAnnotationGenerationFunc(
        namespace,
        interfaces,
        networks,
        networkNameScheme,
        config, 
        func(ns, ifaces, network, podInterfaceName)  {
            return newMultusAnnotationDataWithIPAMClaim(namespace, interfaces, network, podInterfaceName, networkToIPAMClaimParams[network.Name].ClaimName)
        },
    )
}

func commonMultusAnnotationGenerationFunc(namespace string, interfaces []v1.Interface, networks []v1.Network, networkNameScheme map[string]string, config *virtconfig.ClusterConfig, fn func(ns, ifaces, network, podInterfaceName)) (string, error) {
multusNetworkAnnotationPool := multusNetworkAnnotationPool{}

	for _, network := range networks {
		if vmispec.IsSecondaryMultusNetwork(network) {
			podInterfaceName := networkNameScheme[network.Name]
			multusNetworkAnnotationPool.add(
			    fn(namespace, interfaces, network, podInterfaceName),
			)
		}

		if config != nil && config.NetworkBindingPlugingsEnabled() {
			if iface := vmispec.LookupInterfaceByName(interfaces, network.Name); iface.Binding != nil {
				bindingPluginAnnotationData, err := newBindingPluginMultusAnnotationData(
					config.GetConfig(), iface.Binding.Name, namespace, network.Name)
				if err != nil {
					return "", err
				}
				if bindingPluginAnnotationData != nil {
					multusNetworkAnnotationPool.add(*bindingPluginAnnotationData)
				}
			}
		}
	}
	if !multusNetworkAnnotationPool.isEmpty() {
		return multusNetworkAnnotationPool.toString()
	}
	return "", nil
}
```

Reply:
there is only one case where we don't need the new parameter (backward compatibility code `PodHasOrdinalInterfaceName`)
in the rest cases which aren't a lot we do need it.
imho it adds bit complexity over the current suggested code